### DISCUSSION TITLE
Decoupling IB

### DISCUSSION THREAD

**Discussion by nwillemse** [dodid/minitrade#8] [General]
Hi,

Minitrade seems to be exactly what I have been wanting to develop for myself.  I have started looking into it.

I see that IB seems to be tightly integrated with minitrade. However, for my instance I am running IB on a totally different server for instance.  It would be great to have an option to allow to specify the hostname, port, clientid and then use that to connect to the broker.

Any ideas on how to achieve this?

Nick

**Comment by nwillemse** [comment]
I see, you are using the Client Portal Gateway and not the IBGateway...

**Comment by dodid** [comment]
Yes. It's a bit lighter weight and suffice for the project.
